bots/succubus/succubus_main.lua

```diff
@@ -0,0 +1,584 @@
+object.tSkills = {
+	1, 2, 1, 2, 1,
+	3, 1, 2, 2, 0, 
+	3, 0, 0, 0, 4,
+	3, 4, 4, 4, 4,
+	4, 4, 4, 4, 4,
+}
+function object:SkillBuild()
+	core.VerboseLog("skillbuild()")
+
+	local unitSelf = self.core.unitSelf
+	if  skills.smitten == nil then
+		skills.smitten = unitSelf:GetAbility(0)
+		skills.heartache = unitSelf:GetAbility(1)
+		skills.mesme = unitSelf:GetAbility(2)
+		skills.hold = unitSelf:GetAbility(3)
+	end
+	if unitSelf:GetAbilityPointsAvailable() <= 0 then
+		return
+	end
+
+	local nlev = unitSelf:GetLevel()
+	local nlevpts = unitSelf:GetAbilityPointsAvailable()
+	for i = nlev, nlev+nlevpts do
+		unitSelf:GetAbility( object.tSkills[i] ):LevelUp()
+	end
+end
+
+---------------------------------------------
+--	This adds puzzle box to the illusions  --
+---------------------------------------------
+
+function illusionLib.updateIllusions(botBrain)
+	illusionLib.tIllusions = {}
+	local tPossibleIllusions = core.tControllableUnits["AllUnits"]
+	if tPossibleIllusions ~= nil then
+		for nUID, unit in pairs(tPossibleIllusions) do
+			local sTypeName = unit:GetTypeName()
+			if sTypeName ~= "Pet_GroundFamiliar" and sTypeName ~= "Pet_FlyngCourier" then
+				tinsert(illusionLib.tIllusions, unit)
+			end
+		end
+	end
+end
+
+object.retreatCastThreshold = 55
+function behaviorLib.CustomRetreatExecute(botBrain)
+	local unitSelf = core.unitSelf
+	local mypos = unitSelf:GetPosition()
+
+	local lastRetreatUtil = behaviorLib.lastRetreatUtil
+
+	local missingHP = unitSelf:GetMaxHealth() - unitSelf:GetHealth()
+
+	local mesmeRange = skills.mesme:GetRange()
+	local heartacheRange = skills.heartache:GetRange()
+	local mesmeCanActivate = skills.mesme:CanActivate()
+	local heartacheCanActivate = skills.heartache:CanActivate()
+	local bActionTaken = false
+
+	local itemPortalKey = core.GetItem("Item_PortalKey")
+	if lastRetreatUtil > object.retreatCastThreshold and itemPortalKey ~= nil then
+		if itemPortalKey:CanActivate() then
+			botBrain:OrderItemPosition(itemPortalKey.object, behaviorLib.GetSafeBlinkPosition(core.allyWell:GetPosition(), 1200))
+			bActionTaken = true
+		end
+	end
+
+	if not bActionTaken then
+		if lastRetreatUtil > object.retreatCastThreshold then
+			for _, hero in pairs(core.localUnits["EnemyHeroes"]) do
+				if not hero:isMagicImmune() then
+					distanceSq = Vector3.Distance2DSq(mypos, hero:GetPosition())
+					if heartacheCanActivate and distanceSq < heartacheRange * heartacheRange and missingHP > 300 and not hero:HasState("State_Succubis_Ability3") then
+						bActionTaken = core.OrderAbilityEntity(botBrain, skills.heartache, hero)
+						break
+					end
+					if mesmeCanActivate and distanceSq < mesmeRange * mesmeRange then
+						bActionTaken = core.OrderAbilityEntity(botBrain, skills.mesme, hero)
+						break
+					end
+				end
+			end
+		end
+	end
+
+	return bActionTaken
+end
+
+function behaviorLib.CustomReturnToWellExecute(botBrain)
+	local itemPortalKey = core.GetItem("Item_PortalKey")
```

Comment:
This function should simply be:
core.OrderBlinkItemToEscape(botBrain, core.unitSelf, core.GetItem("Item_PortalKey"), true)